src/main/sensors/gyroanalyse.c

```diff
@@ -46,58 +46,85 @@
 // Eg [0,31), [31,62), [62, 93) etc
 
 #define FFT_BIN_COUNT             (FFT_WINDOW_SIZE / 2)
-// compare 1 + this offset FFT bins for peak, ie if 1 start 2.5 * 41.655 or about 104Hz
-#define FFT_BIN_OFFSET            1
-// analyse up to 666Hz, 16 bins each 41.625 Hz wide
+// for gyro loop >= 4KHz, analyse up to 666Hz, 16 bins each 41.625 Hz wide
 #define FFT_SAMPLING_RATE_HZ      1333
-// centre frequency of bandpass that constrains input to FFT
-#define FFT_BPF_HZ                (FFT_SAMPLING_RATE_HZ / 4)
-// Hz per bin
-#define FFT_RESOLUTION            ((float)FFT_SAMPLING_RATE_HZ / FFT_WINDOW_SIZE)
 // following bin must be at least 2 times previous to indicate start of peak
 #define FFT_MIN_BIN_RISE          2
 
 // lowpass frequency for smoothing notch centre point
 #define DYN_NOTCH_SMOOTH_FREQ_HZ  60
 // notch centre point will not go below this, must be greater than cutoff, mid of bottom bin
 #define DYN_NOTCH_MIN_CENTRE_HZ   125
-// maximum notch centre frequency limited by Nyquist
-#define DYN_NOTCH_MAX_CENTRE_HZ   (FFT_SAMPLING_RATE_HZ / 2)
 // lowest allowed notch cutoff frequency
 #define DYN_NOTCH_MIN_CUTOFF_HZ   105
 // we need 4 steps for each axis
 #define DYN_NOTCH_CALC_TICKS      (XYZ_AXIS_COUNT * 4)
 
+static bool FAST_RAM_ZERO_INIT gyroAnalyseInitialized;
+static uint16_t FAST_RAM_ZERO_INIT fftSamplingRateHz;
+// centre frequency of bandpass that constrains input to FFT
+static uint16_t FAST_RAM_ZERO_INIT fftBpfHz;
+// Hz per bin
+static float FAST_RAM_ZERO_INIT    fftResolution;
+// maximum notch centre frequency limited by Nyquist
+static uint16_t FAST_RAM_ZERO_INIT dynNotchMaxCentreHz;
+static uint8_t  FAST_RAM_ZERO_INIT fftBinOffset;
+
 // Hanning window, see https://en.wikipedia.org/wiki/Window_function#Hann_.28Hanning.29_window
 static FAST_RAM_ZERO_INIT float hanningWindow[FFT_WINDOW_SIZE];
 static FAST_RAM_ZERO_INIT float dynamicNotchCutoff;
 
-void gyroDataAnalyseInit(void)
+// Making this NOINLINE saves a few bytes to help F3's compile
+// It's only called once at initialization so no performance concerns
+NOINLINE void gyroDataAnalyseInit(uint32_t targetLooptimeUs)
```

Comment:
I am surprised that the `NOINLINE` should save space if the function is only called once - after all we save one function call. Also, not sure if we want to go down the route of using compiler directives like this to optimise for F3, considering they will affect the other MCU types as well.

Reply:
I am surprised too, should review generated assembly to find out. On targets without dual gyro should not matter.

Reply:
Good point. Making the check for only one initialisation conditional to dual gyro would probably also get us out of the F3 flash space danger zone.

Reply:
Well maybe this `NOINLINE` was a false finding - not sure now.  At the time I added it the code was overflowing OMNIBUS by about 75 bytes.  Adding the `NOINLINE` allowed it to compile.  However trying now with/without I'm only seeing about 4 bytes difference so I can't explain what was happening before.  Possibly other code changes negated whatever effect the `NOINLINE` was having - not sure.

Reply:
Was the overflow happening on local builds or on builds in Travis?

Reply:
Local builds.  But like I said, can't replicate the effect now.